hlapipeline/alignimages.py

```diff
@@ -214,69 +220,61 @@ def perform_align(input_list, archive=False, clobber=False, update_hdr_wcs=False
-    if best_fit_rms > MAX_FIT_RMS:
-        for catalogIndex in range(1, len(catalogList)):
-            print("-------------------- STEP 6: Detect catalog astrometric sources --------------------")
-            print("Astrometric Catalog: ",catalogList[catalogIndex])
-            reference_catalog = generate_astrometric_catalog(processList, catalog=catalogList[catalogIndex])
-
-            if len(reference_catalog) < MIN_CATALOG_THRESHOLD:
-                print("Not enough sources found in catalog " + catalogList[catalogIndex])
-                print("Try again with other catalog")
-                catalogIndex += 1
-                retry_fit = True
-                skip_all_other_steps = True
-            else:
-                print("-------------------- STEP 6b: Cross matching and fitting --------------------")
 
-                fit_rms, fit_num = match_default_fit(imglist, reference_catalog,
-                                     print_fit_parameters=print_fit_parameters)
+    best_fit_rms = -99999.0
+    fit_algorithm_list= ['hack_2d_hist','default']
+    for catalogIndex in range(0, len(catalogList)): #loop over astrometric catalog
+        print("-------------------- STEP 5: Detect astrometric sources --------------------")
+        print("Astrometric Catalog: ",catalogList[catalogIndex])
+        reference_catalog = generate_astrometric_catalog(processList, catalog=catalogList[catalogIndex])
+
+        if len(reference_catalog) < MIN_CATALOG_THRESHOLD:
+            print("Not enough sources found in catalog " + catalogList[catalogIndex])
+            print("Try again with other catalog")
+        else:
+            print("-------------------- STEP 5b: Cross matching and fitting --------------------")
+            for algorithm_name in fit_algorithm_list: #loop over fit algorithm type
+                print("------------------ ------------------ ------------------ {} {} ------------------ ------------------ ------------------".format(catalogList[catalogIndex],algorithm_name))
+                # Convert input images to tweakwcs-compatible NDData objects and
+                # attach source catalogs to them.
+                imglist = []
+                for group_id, image in enumerate(processList):
+                    img = amutils.build_nddata(image, group_id,
+                                               extracted_sources[image]['catalog_table'])
+                    # add the name of the image to the imglist object
+                    for im in img:
+                        im.meta['name'] = image
+                    imglist.extend(img)
+
+                #copy in best-fit solution to newly initialized imglist
+                if best_fit_rms >= 0.:
+                    for item,temp_item in zip(imglist,imglist_temp):
+                        item.best_meta = temp_item.best_meta.copy()
+
+                #execute the correct fitting/matching algorithm
+                if algorithm_name == "hack_2d_hist":
```

Comment:
Since these functions should share a common API (inputs and return values), then having algorithm_name be the function names directly means these 2 if statements can be completely eliminated by simply calling the algorithm_name directly. 

fit_rms, fit_num = algorithm_name(imglist, reference_catalog, print_fit_parameters=print_fit_parameters)

Reply:
Resolved by commit [e039da3](https://github.com/spacetelescope/hlapipeline/pull/57/commits/e039da33ad03cdbf0c739f159f7cc900e9de0984)